src/evict/evict_lru.c

```diff
@@ -1555,18 +1615,6 @@ __evict_walk(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue)
             continue;
         }
 
-        /*
-         * Skip files if we have too many active walks.
-         *
-         * This used to be limited by the configured maximum number of hazard pointers per session.
-         * Even though that ceiling has been removed, we need to test eviction with huge numbers of
-         * active trees before allowing larger numbers of hazard pointers in the walk session.
-         */
-        if (btree->evict_ref == NULL && session->hazards.num_active > WT_EVICT_MAX_TREES) {
-            WT_STAT_CONN_INCR(session, cache_eviction_server_skip_trees_too_many_active_walks);
```

Comment:
It's important to consider some potential downsides of this change of removal of the cap on trees

Previously, only the 1000 best trees were considered for eviction, ensuring a focused process with the session having a limit (1000) on how many pages to pin. However, with the cap removed, the system may now consider (and have a hazard pointer) on up to a many (million?) trees, increasing the time spent on less important trees and potentially leading to inefficiencies (as the cap ensures less important trees are skipped and never walked).

Another downside is the retention of hazard pointers. Even in scenarios where most trees are suitable for eviction, the position is retained throughout the db for each page per tree. This long retention can prevent pages from being evicted, holding the hazard pointers longer than necessary and potentially prevting the same page from getting evicted.

Moreover, I assume that the benefits of these changes are speculative.This uncertainty makes the changes seem risky. 

@ershov is there any test that has proven an improvement in performance from getting rid of this check?